File: LLM-Translations/GPT-4/avatar/Python/C/codeforces_678_B.c

#include <stdio.h>
#include <stdbool.h>

bool leapyear(int n) {
    if (n % 400 == 0 || (n % 4 == 0 && n % 100 != 0)) {
        return true;
    } else {
        return false;
    }
}

int main() {
    int n;
    char d = 'f';
    scanf("%d", &n);

    bool ans;
    if (n % 400 == 0 || (n % 4 == 0 && n % 100 != 0)) {
        ans = true;
    } else {
        ans = false;
    }

    while (1) {
        if (leapyear(n)) {
            if (d == 'm') {
                d = 't';
                n++;
            } else if (d == 't') {
                d = 'w';
                n++;
            } else if (d == 'w') {
                d = 'h';
                n++;
            } else if (d == 'h') {
                d = 'f';
                n++;
            } else if (d == 'f') {
                d = 's';
                n++;
            } else if (d == 's') {
                d = 'm';
                n++;
            }
        } else {
            if (d == 'm') {
                d = 't';
                n++;
            } else if (d == 't') {
                d = 'w';
                n++;
            } else if (d == 'w') {
                d = 'h';
                n++;
            } else if (d == 'h') {
                d = 'f';
                n++;
            } else if (d == 'f') {
                d = 's';
                n++;
            } else if (d == 's') {
                d = 'm';
                n++;
            }
        }

        if (d == 'f' && ans == true && leapyear(n)) {
            break;
        } else if (d == 'f' && ans == false && !leapyear(n)) {
            break;
        }
    }

    printf("%d", n);

    return 0;
}
// End of Code

File: LLM-Translations/GPT-4/avatar/Python/Go/codeforces_678_B.go
package main

import (
    "fmt"
    "bufio"
    "os"
    "strconv"
)

func leapyear(n int) string {
    if n % 400 == 0 || (n % 4 == 0 && n % 100 != 0) {
        return "y"
    } else {
        return "n"
    }
}

func main() {
    reader := bufio.NewReader(os.Stdin)
    input, _ := reader.ReadString('\n')
    n, _ := strconv.Atoi(input[:len(input)-1])
    d := "f"
    var ans string
    if n % 400 == 0 || (n % 4 == 0 && n % 100 != 0) {
        ans = "ly"
    } else {
        ans = "nly"
    }
    for {
        if n % 400 == 0 || (n % 4 == 0 && n % 100 != 0) {
            switch d {
            case "m", "t", "w", "th", "f", "sa", "s":
                d = map[string]string{"m": "t", "t": "w", "w": "th", "th": "f", "f": "sa", "sa": "s", "s": "m"}[d]
                n += 1
            }
        } else {
            switch d {
            case "m", "t", "w", "th", "f", "sa", "s":
                d = map[string]string{"m": "t", "t": "w", "w": "th", "th": "f", "f": "sa", "sa": "s", "s": "m"}[d]
                n += 1
            }
        }
        if (d == "f" && ans == "ly" && leapyear(n) == "y") || (d == "f" && ans == "nly" && leapyear(n) == "n") {
            break
        }
    }
    fmt.Println(n)
}

// End of Code.
